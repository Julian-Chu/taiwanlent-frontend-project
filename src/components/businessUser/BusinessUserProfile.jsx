
import React, { Component } from 'react';
// import regions from '../common/regions';
import { connect } from 'react-redux';
import { Field, reduxForm, reset } from 'redux-form';
// import checkRules from '../../regularExpression/checkRules';
import { GetBusinessUserData, UpdateBusinessUserData } from '../../actions/businessuser';
import validate from './BusinessUserRegisterPages/validate';
import renderField from '../renderComponents/renderField';
import renderRadio from '../renderComponents/renderRadio';
import '../../styles/ReduxForm.css';


export class BusinessUserProfile extends Component {
  constructor(props) {
    super(props);
    this.state = {
      disabled: true
    };

    this.onFormSubmit = this.onFormSubmit.bind(this);
  }

  componentDidMount() {
    // this.props.setHeaderNontransparent();
    this.props.GetBusinessUserData();
  }

  onFormSubmit(values) {
    // console.log(values);
    // console.log('this.props:', this.props);
    // var history = this.props.history;
    // console.log('history: ', history);
    this.props.UpdateBusinessUserData(values, ()=>this.toggleChangeInput(true));
  }

  toggleChangeInput(disabled) {
    this.setState({
      disabled: disabled
    })
  }

  renderButtons(handleSubmit, pristine, submitting) {
    if (this.state.disabled) {
      return (
        <div>
          <button type="button" className="button button-border button-dark button-circle" onClick={() => this.toggleChangeInput(false)}>修改資料</button>
        </div>
      )
    } else {
      return (
        <div>
          {
            !(pristine || submitting) &&
            <button type="button" className="button button-border button-dark button-circle" onClick={handleSubmit(this.onFormSubmit)}>confirm</button>
          }
          <button type="button" className="button button-border button-dark button-circle" onClick={() => {
            this.props.reset('BusinessUserProfileForm');
            this.toggleChangeInput(true);
          }

          }>Cancel</button>

        </div>
      )
    }
  }



  render() {
    const { handleSubmit, pristine, submitting } = this.props;
    return (
      <div className="content-wrap">
        <div className="container clearfix">
          <div className="col_two_third col_last nobottommargin">
            <h1>廠商註冊</h1>
            <form id="register-form" name="register-form" className="nobottommargin" onSubmit={handleSubmit(this.onFormSubmit.bind(this))} disabled="true">
              {/* <Field
                name="username"
                placeholder=""
                className="col_half"
                title="使用者名稱"
                component={renderField}
                disabled="true"
              /> */}
              <div className="col_half">
                <label htmlFor="">使用者名稱</label>
                <div>{this.props.initialValues.username}</div>
              </div>
              <div id="gender">
                <label>性別</label>
                <div>
                  <Field
                    name="gender"
                    required={true}
                    options={[
                      { title: '男', value: 'male' },
                      { title: '女', value: 'female' }
                    ]}
                    component={renderRadio}
                    disabled={this.state.disabled}
                  ></Field>
                </div>
              </div>
              <div className="clear"></div>

              <Field
                name="email"
                placeholder=""
                className="col_half"
                title="Email"
                component={renderField}
                disabled={this.state.disabled}
              />
              {/* <Field
                name="reemail"
                placeholder=""
                className="col_half col_last"
                title="再次確認Email"
                component={renderField}
                disabled={this.state.disabled}
              /> */}
              <div className="clear"></div>
              {/* <Field
                name="password"
                placeholder=""
                className="col_half"
                title="輸入密碼"
                type="password"
                component={renderField}
                disabled={this.state.disabled}
              />
              <Field
                name="repassword"
                placeholder=""
                className="col_half col_last"
                title="再次輸入密碼"
                type="password"
                component={renderField}
                disabled={this.state.disabled}
              />
              <div className="clear"></div> */}

              <Field
                name="name"
                placeholder=""
                className="col_half"
                title="真實姓名"
                component={renderField}
                disabled={this.state.disabled}
              />
              <Field
                name="phone"
                placeholder=""
                className="col_half col_last"
                title="連絡電話"
                component={renderField}
                disabled={this.state.disabled}
              />
              <div className="clear"></div>
              <Field
                name="companyName"
                title="公司名稱/個人委託"
                placeholder=""
                className="col_half "
                disabled={this.state.disabled}
                component={renderField} />

              <Field
                name="department"
                title="部門"
                placeholder=""
                className="col_half col_last"
                disabled={this.state.disabled}
                component={renderField} />
              <div className="clear"></div>

              <Field
                name="companyLocation"
                title="公司所在城市"
                placeholder=""
                className="col_half "
                disabled={this.state.disabled}
                component={renderField} />

              <Field
                name="address"
                title="地址"
                placeholder=""
                className="col_half col_last"
                disabled={this.state.disabled}
                component={renderField} />
              <div className="clear"></div>

              <Field
                name="industry"
                title="產業類別"
                placeholder=""
                className="col_half"
                disabled={this.state.disabled}
                component={renderField} />

              <Field
                name="productIntroduction"
                title="產品簡介"
                placeholder=""
                className="col_half col_last"
                disabled={this.state.disabled}
                component={renderField} />
              <div className="clear"></div>
              <div className="col_half col_last">
                <br />
              </div>

              <div className="clear"></div>
              <div className="col_full nobottommargin">
                {this.renderButtons(handleSubmit, pristine, submitting)}
              </div>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

BusinessUserProfile = reduxForm({
  validate,
  form: 'BusinessUserProfileForm',
  enableReinitialize: true
})(BusinessUserProfile);

const ConnectedBusinessUserProfile = connect(
  state => {
    var initialValues = state.businessUserData;
    return {
      initialValues
    }
  }, { GetBusinessUserData, UpdateBusinessUserData, reset })(BusinessUserProfile)


export default ConnectedBusinessUserProfile;

// export default reduxForm({
//   validate,
//   form: 'BusinessUserRegisterForm'
// })(connect(null, actions)(BusinessUserProfile));