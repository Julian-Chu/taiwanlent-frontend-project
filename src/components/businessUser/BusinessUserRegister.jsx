import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FirstPage from './BusinessUserRegister/FirstPage';
import SecondPage from './BusinessUserRegister/SecondPage';

class BusinessUserRegister extends Component {
  constructor(props) {
    super(props);
    this.nextPage = this.nextPage.bind(this);
    this.previousPage = this.previousPage.bind(this);
    this.renderPageByNumber = this.renderPageByNumber.bind(this);
    this.state = {
      page: 1
    };
  }

  nextPage() {
    this.setState({ page: this.state.page + 1 });
  }
  previousPage() {
    this.setState({ page: this.state.page - 1 });
  }

  renderPageByNumber(page, onSubmit) {
    switch (page) {
      case 1:
        return (
          <FirstPage onSubmit={this.nextPage}></FirstPage>
        )
      case 2:
        return (
          <SecondPage onSubmit={onSubmit} previousPage={this.previousPage}></SecondPage>
        )
    }
  }

  render() {
    const { onSubmit } = this.props;
    const { page } = this.state;
    return (
      <div style={{margin: '5% 10% 5%'}}>
        <h1>廠商註冊</h1>
        {this.renderPageByNumber(page, onSubmit)}
        <ol className="nav">
          <li>
            <button  className='active' >1</button>
          </li>
          <li>
            <button  className={this.state.index >= 1 ? 'active' : ''} >2</button>
          </li>
        </ol><h6>{this.state.page}/2</h6>
      </div>
    )
  }
}

export default BusinessUserRegister;