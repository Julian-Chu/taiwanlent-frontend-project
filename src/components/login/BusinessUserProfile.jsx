import React, {Component} from 'react';
import {connect} from 'react-redux';
import { bindActionCreators } from 'redux';
import {GetBusinessUserData} from '../../actions/businessuser';

class BusinessUserProfile extends Component{
  constructor(props){
    super(props);
    this.props.GetBusinessUserData();
  }
  // componentWillMount(){
  //   this.props.GetBusinessUserData();
  // }
  componentDidMount(){
    console.log(this.props.userdata);
    // this.props.GetBusinessUserData();
    setTimeout(()=>{
      console.log('timer');
    console.log(this.props.userdata);
      
      this.setState({})
    }, 2000)
  }
  renderField(field){
    return(
      <div>
        <label htmlFor="">{field.title}</label>
        <label htmlFor="">{field.value}</label>
      </div>
    )
  }
  render(){
    return(
      <div className="container">
        <h1>Test</h1>
        <div className="col_half">
          <label htmlFor="">使用者名稱:{this.props.userdata.username}</label>
          <br/>
          <label htmlFor="">性別:</label>
          <label htmlFor="">{this.props.userdata.gender}</label>
        </div>

      </div>
    )
  }
}

function mapStateToProps(state){
  console.log(state);
  return {userdata: state.businessUserData}
};

// const mapDispatchToProps = dispatch =>(
//   {
//     GetBusinessUserData:()=>dispatch(businessUserActions.GetBusinessUserData())
//   }
// )
function mapDispatchToProps(dispatch) {
  return bindActionCreators({GetBusinessUserData}, dispatch);
}


export default connect(mapStateToProps,mapDispatchToProps)(BusinessUserProfile);