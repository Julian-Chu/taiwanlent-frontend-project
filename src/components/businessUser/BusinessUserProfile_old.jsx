import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { GetBusinessUserData } from '../../actions/businessuser';
import BusinessUserRegister from './BusinessUserRegister';

class BusinessUserProfile extends Component {
  constructor(props) {
    super(props);
    this.state = {
      modifyData: false
    }
  }
  componentDidMount() {
    this.props.GetBusinessUserData();
  }
  renderField(field) {
    return (
      <div>
        <label htmlFor="">{field.title}</label>
        <label htmlFor="">{field.value}</label>
      </div>
    )
  }

  renderUserData() {
    return (
      <div>
        <div className="col_half">
          <label>使用者名稱:</label>
          <p>{this.props.userdata.username}</p>
          <label>Email:</label>
          <p>{this.props.userdata.email}</p>
          <label>聯絡電話:</label>
          <p>{this.props.userdata.phone}</p>
          <label>部門名稱:</label>
          <p>{this.props.userdata.department}</p>
          <label>公司所在城市:</label>
          <p>{this.props.userdata.companyLocation}</p>
          <label>地址:</label>
          <p>{this.props.userdata.address}</p>
        </div>
        <div className="col_half col_last">
          <label>真實姓名:</label>
          <p>{this.props.userdata.name}</p>

          <label>性別:</label>
          <p>{this.props.userdata.gender}</p>
          <label>公司名稱/個人委託:</label>
          <p>{this.props.userdata.companyName}</p>
          <label>產業類別:</label>
          <p>{this.props.userdata.industry}</p>
          <label>產品簡介:</label>
          <p>{this.props.userdata.productIntroduction}</p>
        </div>
        <button onClick={() => this.setState({ modifyData: true })}>修改資料</button>

      </div>
    )
  }
  render() {
    return (
      <div className="container">
        <h1>用戶資料</h1>
        {this.state.modifyData? <BusinessUserRegister/>: this.renderUserData()}
      </div>

    )
  }
}

function mapStateToProps(state) {
  console.log(state);
  return { userdata: state.businessUserData }
};

// const mapDispatchToProps = dispatch =>(
//   {
//     GetBusinessUserData:()=>dispatch(businessUserActions.GetBusinessUserData())
//   }
// )
function mapDispatchToProps(dispatch) {
  return bindActionCreators({ GetBusinessUserData }, dispatch);
}


export default connect(mapStateToProps, mapDispatchToProps)(BusinessUserProfile);