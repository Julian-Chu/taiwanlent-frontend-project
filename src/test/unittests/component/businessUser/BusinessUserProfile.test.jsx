
import React from 'react';
import { shallow} from 'enzyme';
import { BusinessUserProfile } from '../../../../components/businessUser/BusinessUserProfile';

describe("BusinessUserProfile",()=>{
  const app = shallow(<BusinessUserProfile></BusinessUserProfile>);

  it("check snapshot", ()=>{
    expect(app).toMatchSnapshot();
  })
})