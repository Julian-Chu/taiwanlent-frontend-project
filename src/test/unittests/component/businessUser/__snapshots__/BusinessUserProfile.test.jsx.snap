// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessUserProfile check snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReduxForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "destroyOnUnmount": true,
      "enableReinitialize": true,
      "forceUnregisterOnUnmount": false,
      "form": "BusinessUserProfileForm",
      "getFormState": [Function],
      "initialValues": undefined,
      "keepDirtyOnReinitialize": false,
      "persistentSubmitErrors": false,
      "pure": true,
      "shouldAsyncValidate": [Function],
      "shouldError": [Function],
      "shouldValidate": [Function],
      "shouldWarn": [Function],
      "touchOnBlur": true,
      "touchOnChange": false,
      "updateUnregisteredFields": false,
      "validate": [Function],
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "destroyOnUnmount": true,
        "enableReinitialize": true,
        "forceUnregisterOnUnmount": false,
        "form": "BusinessUserProfileForm",
        "getFormState": [Function],
        "initialValues": undefined,
        "keepDirtyOnReinitialize": false,
        "persistentSubmitErrors": false,
        "pure": true,
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "touchOnBlur": true,
        "touchOnChange": false,
        "updateUnregisteredFields": false,
        "validate": [Function],
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
